server/task_scheduler.go

```diff
@@ -306,3 +311,27 @@ func (s *TaskScheduler) ScheduleIfNeeded(ctx context.Context, task *api.Task) (*
 
 	return updatedTask, nil
 }
+
+func (s *TaskScheduler) isBlockingTaskDone(ctx context.Context, blockingTaskList []string) bool {
+	for _, blockingTaskIDString := range blockingTaskList {
+		blockingTaskID, err := strconv.Atoi(blockingTaskIDString)
+		if err != nil {
+			s.l.Error("failed to convert id string to int", zap.String("id string", blockingTaskIDString), zap.Error(err))
```

Comment:
We may want isBlockingTaskDone to return (bool, error) because the error in line 319 may be a permanent error that we want to fail early without blocking the task execution.